pkg/dockerregistry/server/repositorymiddleware.go

```diff
@@ -124,16 +139,19 @@ func newRepositoryWithClient(
 		return nil, fmt.Errorf("%s is required", DockerRegistryURLEnvVar)
 	}
 
-	pullthrough := false
-	if value, ok := options["pullthrough"]; ok {
-		if b, ok := value.(bool); ok {
-			pullthrough = b
-		}
+	pullthrough := getBoolOption("pullthrough", false, options)
+
+	var acceptschema2 bool
+
+	if os.Getenv(EnforceAcceptSchema2EnvVar) == "" {
+		acceptschema2 = getBoolOption("acceptschema2", false, options)
+	} else {
+		acceptschema2 = true
```

Comment:
> This is the case when AcceptSchema2EnvVar != "" It means someone wants to turn it on.

It can also mean someone wants to turn it off (override a value in a custom config file). You need to check for `AcceptSchema2EnvVar == "true"` as is done with other boolean env vars.